### ISSUE TITLE
chi2/ndf bug

### ISSUE THREAD

**Issue by piberger** [psi46/MoReWeb#78]
Hi,

i looked at the chi2/ndf plots of the linear fit in the Vcal calibration because they looked suspicious and found a bug in /Analyse/TestResultClasses/CMSPixel/QualificationGroup/XrayCalibration/VcalCalibrationModule/VcalCalibrationROC/VcalCalibrationROC.py (master branch)
the real chi2/ndf is not saved as 'value' but as 'sigma' with the same value than the offset:
````
@@ -163,10 +163,10 @@ class TestResult(GeneralTestResult):
                 'Sigma': round(fit.GetParError(0), 3),
             },
             'chi2': {
-                'Value': round(fit.GetParameter(0), 3),
+                'Value': round(fit.GetChisquare() / fit.GetNDF(), 3),
                 'Label': 'Chi2',
                 'Unit': 'per NDF',
-                'Sigma': round(fit.GetChisquare() / fit.GetNDF(), 3),
````
Furthermore:
The errors we assume are probably too small that's why we get large values of chi2/ndf. If instead of the statistical error of the fit, a constant error of 0.4 Vcal (RMS of a reproducibility study I have done) is used for the peak position, the order of magnitude of the chi2/ndf becomes much better. But before using such constant it should be checked by other test centers if they obtain the same result.
In principle I think we should also consider an error of the conversion from gammas to electrons in silicon, but I haven't found any information on this yet and will search for it. If it is very small we can neglect it of course.

**Comment by veloxid** [comment]
See issue #9

**Comment by piberger** [comment]
should be fixed by #79, please someone cross check, #9 remains open
